optaplanner-website-root/content/download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -1656,3 +1656,137 @@ Even though OptaPlanner 8.x has always required JDK 11, this only becomes a hard
 === Rename optaplanner.solver.solve-length to optaplanner.solver.solve.duration
 
 The Micrometer meter `optaplanner.solver.solve-length` has been renamed to `optaplanner.solver.solve.duration` to follow Micrometer naming conventions and make it appear better in a variety of monitoring systems. References to `optaplanner.solver.solve-length.+++*+++` in the monitoring system should be changed to `optaplanner.solver.solve.duration.+++*+++`.
+
+== From 8.13.0.Final to 8.14.0.Final
+
+[.upgrade-recipe-major]
+=== Constraint Streams: use `forEach(...)` instead of `from(...)`
+
+We have deprecated the following methods on the `ConstraintFactory` interface:
+
+* `from(...)` in favor of `forEach(...)`,
+* `fromUnfiltered(...)` in favor of `forEachIncludingNullVars(...)`,
+* `fromUniquePair(...)` in favor of `forEachUniquePair(...)`.
+
+The new methods behave differently when it comes to nullable planning variables.
```

Comment:
For clarity, I'd state it like this, to avoid anyone not explicitly doing that thinking this might apply to them:
```suggestion
The new methods behave differently when it comes to `nullable=true` planning variables.
```